src/platform/common/crypto.ts

```diff
@@ -14,12 +14,12 @@ import * as hashjs from 'hash.js';
  */
 @injectable()
 export class CryptoUtils implements ICryptoUtils {
-    public createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): string {
+    public async createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): Promise<string> {
         return computeHash(data, algorithm);
     }
 }
 
-export function computeHash(data: string, algorithm: 'SHA512' | 'SHA256' | 'SHA1') {
+export async function computeHash(data: string, algorithm: 'SHA512' | 'SHA256' | 'SHA1') {
```

Comment:
This is the other change, changed from sync to async.
This change was supposed to be in another PR, but was accidentally meged into this, now we have 74 file changes here.